### DISCUSSION TITLE
4.3.1[실습코드 4-7] p.252 validateAndGetUserId 메서드 질문이요!

### DISCUSSION THREAD

**Discussion by MJJ32** [fsoftwareengineer/todo-application#81] [질문과 답변] [answered]
validateAndGetUserId 메서드에서 세번째 주석에 
//위조되지 않았다면 페이로드(Claims) 리턴, 위조라면 예외를 날림
이라고 써있는데 어느부분에서 예외가 나가는 거죠 ??

위조 됐을 경우 null을 반환해서
JwtAuthenticationFilter 에서 예외를 발생시키는 것인가요 ?

**Comment by fsoftwareengineer** [comment, marked as answer]
TokenProvider 내부에 라이브러리를 이용해 검사하는 부분에서 예외가 납니다. 해당 에러는 JwrAuthenticationFilter의 exception catch부분에서 잡습니다. SecurityContext가 설정되지 앉았으므로 이후 다른 기본 핸들러에 의해 403 http status code를 리턴하게 됩니다.
